apps/gallery/test/marionette/fullscreen_image_test.js

```diff
@@ -99,4 +99,50 @@ marionette('the gallery', function() {
       fullscreen_view.fullscreenFrame3);
     assert.strictEqual(translateX, 0);
   });
+
+  test('Change orientation of displayed image in fullscreen', function() {
+    app.tapFirstThumbnail();
+    client.waitFor(function() {
+      return fullscreen_view.displayed;
+    });
+
+    var screen_width = fullscreen_view.screenWidth;
+    var screen_height = fullscreen_view.screenHeight;
+
+    // Checks the image blob is currently being displayed
+    assert.ok(fullscreen_view.hasSrcImageBlobURL(Gallery.ORIGIN,
+      fullscreen_view.displayedImage));
+
+    // Check that there are 5 options displayed beneath the picture
+    assert.equal(fullscreen_view.toolBarOptions.length, 5);
+
+    // Verify that the screen orientation is in portrait mode
+    assert.equal(fullscreen_view.screenOrientation, 'portrait');
+    assert.ok(fullscreen_view.fullScreenToolBar.displayed);
+    assert.equal(fullscreen_view.toolBarWidth, screen_width);
+
+    // Change the screen orientation to landscape mode and verify that
+    // the screen is in landscape mode
+    fullscreen_view.changeOrientation('landscape');
+    assert.equal(fullscreen_view.screenOrientation, 'landscape');
```

Comment:
The python test you're porting sleeps for a second here. The comment says it is only for visual feedback, but orientation changes take a while to propagate... There are resize events and so on, so maybe it would be a good idea to wait 300ms or something before doing these tests, just in case to avoid potential intermittents